allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/results/AllureResultsHack.kt

```diff
@@ -0,0 +1,57 @@
+package com.kaspersky.components.alluresupport.results
+
+import androidx.test.uiautomator.UiDevice
+import com.kaspersky.components.alluresupport.files.attachVideoToAllureReport
+import com.kaspersky.components.alluresupport.files.dirs.AllureDirsProvider
+import com.kaspersky.components.alluresupport.files.resources.AllureResourcesRootDirsProvider
+import com.kaspersky.components.alluresupport.files.resources.impl.AllureResourceFilesProvider
+import com.kaspersky.kaspresso.runner.listener.KaspressoRunListener
+import io.qameta.allure.kotlin.Allure
+import org.junit.runner.Result
+import java.io.File
+
+class AllureResultsHack(
```

Comment:
All the trick is in one place. It lives as long as the instrumentation lives. Stores the stub-to-real videos bindings and injects them into allure results when all the tests are finished. Files are being moved and copied only once.

Being a `KaspressoRunListener` it is stored in our `KaspressoRunner`.